bundles/org.openhab.binding.visualcrossing/src/main/java/org/openhab/binding/visualcrossing/internal/api/rest/HttpClient.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.visualcrossing.internal.api.rest;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpResponseException;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingApiException;
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingAuthException;
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingRateException;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class HttpClient implements RestClient {
+    private static final int TIMEOUT = 10;
+    private static final int IDLE_TIMEOUT = TIMEOUT;
+    private final org.eclipse.jetty.client.HttpClient client;
+
+    public HttpClient(org.eclipse.jetty.client.HttpClient client) {
+        this.client = requireNonNull(client, "client");
+        if (this.client.isStopped()) {
+            throw new IllegalStateException("HttpClient is stopped");
+        }
+    }
+
+    @Override
+    public @Nullable String get(String url, @Nullable Header... headers)
+            throws VisualCrossingApiException, VisualCrossingAuthException, VisualCrossingRateException {
+        var request = requireNonNull(client.newRequest(url));
```

Comment:
See previous comment, also check others, i don;t think these requireNonNull are needed at all. Or are the annotations wrong?